lib/cheers_leader.rb

```diff
@@ -1,21 +1,31 @@
 class CheerLeaderList
+  attr_reader :list
+  
   def initialize(list)
+    raise ArgumentError if !list.is_a?(Array) || list.any? {|x| x.any? { |i| i.nil? } }
     @list = list
+    sort_by_rank # Why this not WORKING ?? 
+  end
+
+  def sort_by_rank(asc = true)
+    asc ? list.sort_by {|x| x[:rank] } : list.sort_by {|x| -x[:rank] }
+  end
+
+  def sort_by_name(asc = true)
+    #  WHY -x[1] NOT WORKING here ? 
+    asc ? list.sort_by {|x| x[:name] } : list.sort {|x, y| y[:name] <=> x[:name] }
```

Comment:
is it supposed to be `x.name` or `x[:name]`?

Reply:
It supposed to be x[:name]
And both is same I believe ! x.name and x[:name]